core/src/main/scala/org/genivi/sota/core/campaigns/CampaignLauncher.scala

```diff
@@ -35,8 +40,20 @@ object CampaignLauncher {
     Future.successful(devices.map(_ -> Set(pkg.id)).toMap)
   }
 
-  def launch (deviceRegistry: DeviceRegistry, updateService: UpdateService, id: Campaign.Id, lc: LaunchCampaign)
-             (implicit db: Database, system: ActorSystem, ec: ExecutionContext): Future[List[Uuid]] = {
+  def sendMsg(namespace: Namespace, devices: Set[Uuid], pkgId: PackageId, updateId: Uuid,
+              messageBus: MessageBusPublisher)
+             (implicit db: Database, system: ActorSystem, ec: ExecutionContext)
+      : Future[Unit] =
+      for {
```

Comment:
Funny, Daniel said the exact same thing, and showed me some examples. I'll add it, seems like a good opportunity to learn about it too.